### PR TITLE
Refactor to use HBaseTableAdmin for metrics tables, addresses: CDAP-1458

### PR DESCRIPTION
**Pull request by abaranau**
https://issues.cask.co/browse/CDAP-1458

### CODE REVIEW

File: cdap-data-fabric/src/main/java/co/cask/cdap/data2/dataset2/lib/table/hbase/HBaseTableAdmin.java [hunk 1]
```diff
@@ -83,10 +84,20 @@ public void create() throws IOException {
     tableDescriptor.addFamily(columnDescriptor);
 
     // if the dataset is configured for readless increments, the set the table property to support upgrades
-    if (supportsReadlessIncrements(spec)) {
+    boolean supportsReadlessIncrements = supportsReadlessIncrements(spec);
+    if (supportsReadlessIncrements) {
       tableDescriptor.setValue(Table.PROPERTY_READLESS_INCREMENT, "true");
     }
 
+    // if the dataset is configured to be non-transactional, the set the table property to support upgrades
```
Comment (gsps1) [inline cdap-data-fabric/src/main/java/co/cask/cdap/data2/dataset2/lib/table/hbase/HBaseTableAdmin.java]: upgrades?

File: cdap-master/src/main/java/co/cask/cdap/data/tools/Main.java [hunk 2]
```diff
@@ -243,12 +243,15 @@ private static void upgradeUserTables(final Injector injector) throws Exception
 
         final boolean supportsIncrement =
           "true".equalsIgnoreCase(desc.getValue(Table.PROPERTY_READLESS_INCREMENT));
+        final boolean transactional =
+          !"true".equalsIgnoreCase(desc.getValue(Constants.Dataset.TABLE_TX_DISABLED));
```
Comment (gsps1) [inline cdap-master/src/main/java/co/cask/cdap/data/tools/Main.java]: if Constants.Dataset.TABLE_TX_DISABLED is true, then its non-transactional right ?

**Comment by abaranau** [comment]
http://builds.cask.co/browse/CDAP-DUT881-1 GREEN

File: cdap-data-fabric/src/main/java/co/cask/cdap/data2/dataset2/lib/table/hbase/HBaseTableAdmin.java (hunk 1 above)
Comment (abaranau) [inline cdap-data-fabric/src/main/java/co/cask/cdap/data2/dataset2/lib/table/hbase/HBaseTableAdmin.java]: yes. We use this property in upgrades. Same as the one above it

**Comment by abaranau** [comment]
tested on distributed cluster. Upgrade updated CPs

File: cdap-data-fabric/src/main/java/co/cask/cdap/data2/dataset2/lib/table/hbase/HBaseTableAdmin.java (hunk 1 above)
Comment (gsps1) [inline cdap-data-fabric/src/main/java/co/cask/cdap/data2/dataset2/lib/table/hbase/HBaseTableAdmin.java]: ok

**Comment by gsps1** [comment]
LGTM :+1:

**Comment by gsps1** [comment]
one question : we don't check TABLE_TX_DISABLED in the upgrade method ?

**Pull request merged**